packages/components/pie-button/src/index.ts

```diff
@@ -32,12 +34,19 @@ export class PieButton extends LitElement {
     @property({ type: Boolean })
     public disabled = false;
 
+    @property({ type: Boolean, reflect: true })
+    public isLoading = false;
+
     @property({ type: Boolean })
     public isFullWidth = false;
 
+    private renderLoadingSpinner () {
```

Comment:
You could (not fully necessary tho) use the `TemplateResult` type check here from Lit since we are specifying a return value type of template.

Reply:
I've removed this now – after speaking to design, they wanted to keep the height/width of the button dependent on the text in the button, so rather than rendering the loading spinner as markup, I changed it to add it as a pseudo-element (so this isn't needed now)